docs/cpp/move-constructors-and-move-assignment-operators-cpp.md

```diff
@@ -125,14 +125,20 @@ The following procedures describe how to write a move constructor and a move ass
     }
     ```
 
-1. In the move assignment operator, add a conditional statement that performs no operation if you try to assign the object to itself.
+1. In the move assignment operator, we need to keep the self-assignment safe, and the simple way is to add a judgment directly.
 
     ```cpp
     if (this != &other)
     {
     }
     ```
 
+    There are many other ways, too, such as `copy-and-swap`.
```

Comment:
Good to provide an example.

Reply:
[CppCoreGuidelines](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#c65-make-move-assignment-safe-for-self-assignment).

In short, I hope that the description is not so "**absolute**", we should analyze the specific problem.

We don't have to list too many other ways to write it, but in the words
Try to say a little that "**there are other ways**" is still needed.